gravitee-apim-rest-api/gravitee-apim-rest-api-services/gravitee-apim-rest-api-services-dynamic-properties/src/main/java/io/gravitee/rest/api/services/dynamicproperties/DynamicPropertiesService.java

```diff
@@ -96,17 +99,31 @@ public void onEvent(Event<ApiEvent, ApiEntity> event) {
 
         switch (event.type()) {
             case DEPLOY:
-                startDynamicProperties(api);
+                if (hasServiceEnabled(api)) {
+                    startDynamicProperties(api);
+                }
                 break;
             case UNDEPLOY:
                 stopDynamicProperties(api);
                 break;
             case UPDATE:
-                update(api);
+                if (hasServiceEnabled(api)) {
+                    update(api);
+                } else {
+                    stopDynamicProperties(api);
+                }
```

Comment:
I would prefer updating the `update` method to manage the case where the service is disabled. (l.130 to l.147)

Reply:
✅